cmd/cmd.go

```diff
@@ -52,21 +61,35 @@ func Run() int {
 		// TODO(komish): Change output streams
 		configExists = false
 	}
-	if len(os.Args[1:]) < 1 && !configExists {
-		fmt.Printf("Not enough arguments provided and the config file ")
-		fmt.Printf("at path %s was not parsed successfully\n(Error: %s).\n\n", configPath, err)
+	if len(args) < 1 {
+		if showHelp { // Want to see if user wants help first before throwing errors at them
+			PrintHelp()
+			return 0
+		} else if !configExists {
+			fmt.Printf("Not enough arguments provided and the config file ")
+			fmt.Printf("at path %s was not parsed successfully\n(Error: %s).\n\n", configPath, err)
+			PrintHelp()
+			return 15
+		} else {
+			fmt.Printf("Not enough arguments provided")
+			PrintHelp()
+			return 15
+		}
+	}
+
+	// Check if User wants help before we go any further
+	if strings.ToLower(args[1]) == "help" {
 		PrintHelp()
-		return 15
+		return 0
 	}
```

Comment:
Consistent with the previous comment to move the "check for help" tasks higher up in the `Run()` function, let's go ahead and move this further up as well. Also, this only check the first argument, we should probably iterate through all arguments looking for the `help` request.

Reply:
I have a mock up of a function for this iteration if you need any assistance with it.

Reply:
For this small CLI application, I was thinking iterating over the args was overkill. You could do more with this though, GoLang expects the help keyword first then the command `go help mod` and `go mod help` doesn't work. I'll gladly make it to your liking as this is your project just mentioning if you want a full-blown CLI with many commands, I suggest using Cobra.

https://github.com/spf13/cobra

I am a stickler for not pulling in too many dependencies into projects and only pull in when I think it is better to have the code maintained by someone else as it distracts from my goal. The only reason I didn't pull in cobra is I only saw a single command in here and pulled in only the flag parsing items cobra uses (POSIX flag parsing is hard to get right).

Reply:
> I am a stickler for not pulling in too many dependencies into projects and only pull in when I think it is better to have the code maintained by someone else as it distracts from my goal

I agree 100% with this. My gut reaction was to request the use of `flags` but made the exception for `pflags` that you used for the exact reason you state.

Currently `unstaged` only accepts `--help` to get us started, but I have a few others in the planning stages that will come in the future. Right now, I only see *maybe* 3 or 4 at most. And they're mutually exclusive and shouldn't require sub-parsing. That puts them in the "not complicated enough to use something like cobra" bucket in my book (at least for now).

As for adding the iteration, I generally like being able to throw it in anywhere amongst positional arguments. I find it helps me not have to clear my current command line to get help. As an example, `--help` and `-h` probably do this now with the way you've implemented it. I could be 5 positional arguments into my command and throw a `--help` in, I would expect the command to ignore all else and print the help. 

It's probably a bad habit on my part 😆 but it can't hurt.

Reply:
@komish I see your point. Well, pushed up changes.